woocommerce/woocommerce-gateway-stripe
# UPE: Port CC happy flow from WCPay

# Changes proposed in this Pull Request:

Issue: #1708

This PR ports the `process_payment` method for the UPE gateway from WC Pay, and adds the `processCheckout` method to the WC Stripe API wrapper.

# Testing instructions

- Enable UPE feature flag:
```sql
INSERT INTO `wp_options` (`option_id`, `option_name`, `option_value`, `autoload`) VALUES (NULL, '_wcstripe_feature_upe', 'yes', 'no') ON DUPLICATE KEY UPDATE `option_value`=VALUES(`option_value`);
```

- Once the feature flag is enabled, you will see the option to toggle UPE on and off.

- Run npm run start:webpack
- Add any item to the cart and go to http://localhost:8082/checkout/
- Select Stripe, and check that the Credit Card form loads correctly
- Use a normal test card (ex: 4242 4242 4242 4242) to complete the checkout
- Verify that the order successfully completes (also verify the payment in the Stripe dashboard)
- Repeat the process with a card that requires authentication (ex: 4000 0027 6000 3184)
- Verify that 3DS auth modal is shown
- Verify that the order successfully completes (also verify the payment in the Stripe dashboard)
